resource-directory/service/deviceMetadataProjection.go

```diff
@@ -10,24 +12,45 @@ import (
 
 // protected by lock in Projection struct in resource-aggregate/cqrs/eventstore/projection.go
 type deviceMetadataProjection struct {
-	data *events.DeviceMetadataSnapshotTaken
+	deviceID string
+
+	private struct {
```

Comment:
Is this a Go v1.18 new feature? I wasn't aware that `private` keyword even existed in golang!